cmd/cloud_sql_proxy/internal/healthcheck/healthcheck.go

```diff
@@ -62,6 +63,17 @@ func NewServer(c *proxy.Client, port string) (*Server, error) {
 		srv:  srv,
 	}
 
+	mux.HandleFunc(startupPath, func(w http.ResponseWriter, _ *http.Request) {
+		select {
+		case <- hcServer.started: // When the channel is closed, the proxy has finished starting up.
+			w.WriteHeader(http.StatusOK)
+			w.Write([]byte("ok"))
+		default:
+			w.WriteHeader(http.StatusServiceUnavailable)
+			w.Write([]byte("error"))
+		}
```

Comment:
nit: Now that we are doing this logic in two places, it might make sense to have a helper function called something like `hcServer.proxyStarted` that returns a bool